transcom/mymove
# B-19036-INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19036)

## Summary

This gives Services Counselors the ability to create moves. To bridge the gap betwen SCs creating a profile, and then creating a move for that profile, we will need to refactor:
- SC search component to display users as well as moves
- Refactor the CreateMoveCustomerInfo and ServicesCounselingAddOrders components to not use the BASE_COUNSELING_MOVE_PATH
- MoveDetails to include the ability to upload orders files

Because of this, I left some component testing lean.

Move goes directly into "NEEDS_SERVICES_COUNSELING" status at the request of the PO

### How to test

1. Login as a services counselor and click Search
2. Search for an existing customer
3. Click "Create New Move" and get directed to the Customer info screen
  `src/components/Table/SearchResultsTable.jsx (button)`
  `src/pages/Office/CreateMoveCustomerInfo/CreateMoveCustomerInfo.jsx (Customer info screen)`

4. Update customer info and take note of updates and save
5. Find yourself at the new Add Orders page. Fill out orders information, take note of it and click next
  `src/pages/Office/ServicesCounselingAddOrders/ServicesCounselingAddOrders.jsx (add orders screen)`
  `src/components/Office/AddOrdersForm/AddOrdersForm.jsx (add orders screen)`
  `src/services/ghcApi.js clicking next fires counselingCreateOrder`
  `pkg/handlers/ghcapi/orders.go to create orders/move/entitlement`
  `pkg/models/move.go to set status to passed in status (this allows SC created moves to go directly to a status where SCs can edit/add shipments)`

6. Find yourself at the move details view of the new move. Scroll down and confirm the orders information and customer information you entered is correct
